src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -1195,7 +1195,7 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
                 AddSynthesizedAttribute(ref attributes, moduleBuilder.SynthesizeNullableAttribute(this, type));
             }
 
-            if (this.ReturnsByRefReadonly)
+            if (this.ReturnsByRefReadonly || HasReadOnlyModifier)
```

Comment:
This looks wrong. Won't this change cause 

``readonly ref Widget Property => ...```

To be emitted the same as 

```ref readonly Widget Proprety => ```

Reply:
In general why should we be emitting the atribute on a property? IT's the method, get or set, which matters here. 

---
In reply to: [267481347](https://github.com/dotnet/roslyn/pull/34260#discussion_r267481347) [](ancestors = 267481347)

Reply:
This is about ensuring that the `IsReadOnlyAttribute` exists regardless of whether the attribute is applied to the return type or to the method.

Reply:
Looks like I mixed up the method that ensures the attribute exists with the method that applies the attribute?

Reply:
Believe so. I can't see a reason why we'd ever emit the `IsReadOnly` attribute onto a property for the `readonly` member feature. Hence any change to the emit phase around attributes is suspect. Sure it should be on the related accessor but that should be covered by the changes in `Source*MethodSymbol`. 

Think we should definitely have a test where we verify the followin property does not have the `IsReadOnly` attribute: `readonly Widget Property => ...`


---
In reply to: [267483003](https://github.com/dotnet/roslyn/pull/34260#discussion_r267483003) [](ancestors = 267483003)

Reply:
Done